### PR TITLE
Full Blob <-> Blob OAuth support

### PR DESCRIPTION
**Pull request by adreed-msft**
Should a user have access to Blob Data Storage Owner, they can generate User Delegation Keys.

What this means for the user is that they can perform service-to-service transfers without generating a SAS token (assuming it's just blob to blob.)

### CODE REVIEW

File: cmd/helpMessages.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ Copies source data to a destination location. The supported directions are:
   - local <-> Azure Blob (SAS or OAuth authentication)
   - local <-> Azure File (Share/directory SAS authentication)
   - local <-> ADLS Gen 2 (SAS, OAuth, or SharedKey authentication)
-  - Azure Blob (SAS or public) <-> Azure Blob (SAS or OAuth authentication)
+  - Azure Blob (SAS, Public, or OAuth) <-> Azure Blob (SAS or OAuth authentication)
```
Comment (mahilleb-msft) [inline cmd/helpMessages.go]: I guess this arrow shouldn't be bidirectional, independent of your change?

```suggestion
  - Azure Blob (SAS, Public, or OAuth) -> Azure Blob (SAS or OAuth authentication)
```

Comment (adreed-msft) [reply cmd/helpMessages.go]: Ah, yeah, that would make sense.

**Comment by adreed-msft** [comment]
Note: This will have to be redone after the refactor, since it lives in the enumerators. Perhaps I should move these steps to cooking instead?

**Comment by JohnRusk** [comment]
@adreed-msft I'm not close enough to the problem to comment on whether you should move these steps.  It sounds like we should defer further action on this PR until the refactor has reached a suitable point, and you have make the necessary changes here.  Is that correct?  I'm going to mark this with a pr-on-hold tag to represent that fact, but please remove that tag if I'm incorrect.

**Comment by adreed-msft** [comment]
That is correct. We should hold this until the refactor reaches a suitable place to continue.

**Comment by adreed-msft** [comment]
Abandoning this PR as a new one is in the works and this is based on old dead code.

**Pull request closed without merging**